### DISCUSSION TITLE
How to get an HOURLY Time Table data in Bravo

### DISCUSSION THREAD

**Discussion by tidypy** [sql-bi/Bravo#489] [Q&A] [answered]
REQUEST:  (Hourly feature)
I’m working with CREATED-DATETIME, and MODIFIED-DATETIME:  [HOURLY]
SOFTWARE = BRAVO V.0.9.6

![image](https://user-images.githubusercontent.com/50213641/191284229-171c7749-0625-4b94-99ed-9ba0adcec57f.png)

Where is the HOURLY Feature?  
![image](https://user-images.githubusercontent.com/50213641/191284353-f02e15db-3967-4ea0-b3b6-48680f13030b.png)

I tried 'Target Measures'  no hourly measures here either. 
![image](https://user-images.githubusercontent.com/50213641/191284582-330ddc2c-6ad2-4d3f-a529-99213cacfbd6.png)

OK SO I’LL MAKE ONE:   Nope.
![image](https://user-images.githubusercontent.com/50213641/191284666-572923f5-be00-4598-aee0-0c45c0a9f321.png)

I'll just R-click on a column and use PBI, this is far too out of scope.  
![image](https://user-images.githubusercontent.com/50213641/191284774-90106b13-9e14-4fbd-9d5c-5edd5dd8bb9a.png)

WHAT ABOUT EXPRESSION EDITING?   ALSO NOPE.  
(Hidden tables; mysterious CALC , VAR, and IF …) 
![image](https://user-images.githubusercontent.com/50213641/191285010-49837387-d871-4c4e-a3c8-239b25b583f8.png)
![image](https://user-images.githubusercontent.com/50213641/191285048-54411394-3600-49b8-b277-64e751b8188f.png)

OK THIRD TRY:  NOW THAT ITS CREATED BY BRAVO;  CAN I TRANSFORM THE COLUMN?  
NOPE.  
![image](https://user-images.githubusercontent.com/50213641/191285123-1a2c7594-eb2c-423b-a0b1-8561b12d4075.png)

![image](https://user-images.githubusercontent.com/50213641/191285160-c3c0973f-e43e-40cf-a103-898fb4d21509.png)

Nope;  there are hidden tables the code is too sensitive.  
![image](https://user-images.githubusercontent.com/50213641/191285290-566c261a-3699-44e9-ba15-e80711f8902a.png)

Any Suggestions?

**Comment by marcosqlbi** [comment, marked as answer]
In order to use Time Intelligence functions, you should keep a Date table with only one row per day. The best practice is to move the hour (or the time within a day) into another dimension, splitting the date and time into two columns in your fact tables. 

https://www.sqlbi.com/articles/optimizing-high-cardinality-columns-in-vertipaq/
https://www.sqlbi.com/blog/alberto/2019/03/25/using-dax-with-datetime-values/
https://www.sqlbi.com/articles/automatic-time-intelligence-in-power-bi/

Bravo implements the same patterns described here: https://www.daxpatterns.com/time-patterns/

I would not suggest you add a column to your Date table, because it would break the time intelligence calculations. However, if you want to do that, the way to go is by creating a customized template as described in the documentation (work in progress, we'll also add videos later, we are working to complete that before v1): https://docs.sqlbi.com/bravo/features/manage-dates/customize-date-template

In the future (after v1) we might consider adding additional templates for a separate Time table (just hours, hours/minutes, ...), but it's not planned for v1 for sure.

Thanks!
